contracts/managers/ACfManager.sol

```diff
@@ -140,16 +140,8 @@ abstract contract ACfManager is IVersioned, IACfManager {
         IERC20 assetERC20 = _assetERC20();
         uint256 tokensSold = state.totalTokensSold;
         uint256 tokensRefund = assetERC20.balanceOf(address(this)) - tokensSold;
-        IAssetCommon(state.asset).finalizeSale();
-        if (fundsRaised > 0) {
-            (address treasury, uint256 fee) = _calculateFee();
-            if (fee > 0 && treasury != address(0)) {
-                sc.safeTransfer(treasury, fee);
-                sc.safeTransfer(msg.sender, fundsRaised - fee);
-            } else {
-                sc.safeTransfer(msg.sender, fundsRaised);
-            }
-        }
+        _safeFinalizeSale();
+        _safeDistributeFunds(msg.sender, fundsRaised, sc);
```

Comment:
Why is a safe transfer needed?

Reply:
If for any reason FeeManager interaction fails we still want to allow the campaign manager to finalize the campaign , or else he's locked out of his funds. In other words, we want the fee calculation to silently fail